commit c9e3b8e43dcc18560f5d5572f2dfbcfb779656c9
Author: haolong.zhang <haolong.zhang@ck-telecom.com>
Date:   Wed Oct 8 14:33:56 2014 +0800

    [BugFix][CKT AT COMMADN] update enable/disable charge function: sync status and cmd_discharge
    
    [id]:66528
    [Type]
       BUG
    
    [Products]
    	hicam-baseline
    [Requirement]
       N/A
    
    [Solution Description]
       N/A
    
    [Other Info]
       N/A
    
     要提交的变更：
    	修改:         mediatek/kernel/drivers/power/battery_common.c
    
    Change-Id: Ib1613c90c197a7701e67595f2a6c8dfab23c00d0

diff --git a/mediatek/kernel/drivers/power/battery_common.c b/mediatek/kernel/drivers/power/battery_common.c
index 6c180b2..c054f66 100644
--- a/mediatek/kernel/drivers/power/battery_common.c
+++ b/mediatek/kernel/drivers/power/battery_common.c
@@ -550,13 +550,19 @@ static int battery_set_property(struct power_supply *psy,
 		if(val->intval == 1)
 		{
 			charging_enable = true;
+			data->BAT_STATUS = POWER_SUPPLY_STATUS_CHARGING;
+			cmd_discharging = 0;
 			battery_charging_control(CHARGING_CMD_ENABLE,&charging_enable);
+			wake_up_bat();  //sync UI
 			battery_xlog_printk(BAT_LOG_CRTI, "----%s:[%d]:enable charge ok\n", __func__, __LINE__);
 		}
 		else if(val->intval == 0)
 		{
 			charging_enable = false;
+			data->BAT_STATUS = POWER_SUPPLY_STATUS_NOT_CHARGING;
+			cmd_discharging = 1;
 			battery_charging_control(CHARGING_CMD_ENABLE,&charging_enable);
+			wake_up_bat(); //sync UI
 			battery_xlog_printk(BAT_LOG_CRTI, "----%s:[%d]:disable charge ok\n", __func__, __LINE__);
 		}
 		break;
